codec-http/src/main/java/io/netty/handler/codec/http/cookie/ServerCookieEncoder.java

```diff
@@ -163,17 +191,32 @@ public String encode(Cookie cookie) {
      * @return the corresponding bunch of Set-Cookie headers
      */
     public List<String> encode(Iterable<? extends Cookie> cookies) {
-        if (!checkNotNull(cookies, "cookies").iterator().hasNext()) {
+        Iterator<? extends Cookie> cookiesIt = checkNotNull(cookies, "cookies").iterator();
+        if (!cookiesIt.hasNext()) {
             return Collections.emptyList();
         }
 
-        List<String> encoded = new ArrayList<String>();
-        for (Cookie c : cookies) {
-            if (c == null) {
-                break;
+        if (strict) {
+            Map<String, Cookie> nameToCookie = new HashMap<>();
+            while (cookiesIt.hasNext()) {
+                Cookie c = cookiesIt.next();
+                if (c == null) {
+                    break;
+                }
+                nameToCookie.put(c.name(), encode(c));
+            }            
+            return new ArrayList<>(nameToCookie.values());
```

Comment:
Yes, the order is preserved in a newer commit. Thanks. However, you are right that the order doesn't matter, since the returned values will each be written to separate HTTP headers, and the header order is not important.

Unit tests would have to be modified to be order-invariant, and it's possible that the unexpected order may cause user consternation, but see if the efficiency overhead is worth it in the newer commit (which preserves order).